### ISSUE TITLE
New rule: Nullable val can be non-nullable

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#4376]
## Expected Behavior of the rule

Detect variables and properties that are declared as Nullable but they never contain a `null` value.

Non-compilant:
```kotlin
val a: String? = "hello"

val a: String?
   get() = "hello"
```

## Context
The tanslator of code from java to kotlin adds a lot of unnecesary `?`. And when you are refactoring maybe changes a property that could hold `null` to something that couldn't hold `null` but you can forget about redeclaring the type.

I don't know if we could handle `var` too.

After reading the last commits of #4353 I relasied that this rule shouldn't be that difficult to implement.

**Comment by severn-everett** [comment]
I think the case where `var` could be flagged is if the analyzer could prove that any values that it gets assigned are non-null (or at least not nullable). Private `var`s - along with `vars` that have private setters - seen pretty feasible for this; `var`s that are settable outside of their containing object/class might balloon the complexity (and/or computational time) quite a bit.

**Comment by cortinico** [comment]
Closing this as it was shipped in #4379
